src/dispatcher.cc

```diff
@@ -164,55 +185,25 @@ int Dispatcher::Getaddrinfo(const char* node, const char* service, const addrinf
 }
 
 int Dispatcher::Accept4(int sockfd, sockaddr* addr, socklen_t* addrlen, int flags) {
-  sockaddr sock_addr{};
-  socklen_t sock_len{};
-
-  const int fd = raw_->Accept4(sockfd, &sock_addr, &sock_len, flags);
-
-  if (fd < 0) {
-    return fd;
-  }
-  if (addr != nullptr && addrlen != nullptr) {
-    *addr = sock_addr;
-    *addrlen = sock_len;
-  }
-
-  // 1. parse IP out of sockaddr
-  std::string ip_buf(NI_MAXHOST, ' ');
-  std::string port_buf(NI_MAXSERV, ' ');
-  int ret = getnameinfo(&sock_addr, sock_len, ip_buf.data(), NI_MAXHOST, port_buf.data(), NI_MAXSERV, NI_NUMERICHOST | NI_NUMERICSERV);
-  if (ret != 0) {
-    return -1;
-  }
-
-  ip_buf.erase(ip_buf.find('\0'));
-  port_buf.erase(port_buf.find('\0'));
-  std::string domain_port = ip_buf + ":" + port_buf;
-  std::string hostname;
-  // prefer domains over IPs
+  std::string entry_name;
   {
-    const std::lock_guard<std::mutex> lock(domain_mtx_);
-    const auto it = ip_domain_.find(ip_buf);
-    if (it != ip_domain_.cend()) {
-      hostname = it->second;
-      domain_port = hostname + ":" + port_buf;
+    std::lock_guard<std::mutex> lock(tls_fds_mtx_);
+    const auto& entry = fd_entry_.find(sockfd);
+    if (entry == fd_entry_.cend()) {
```

Comment:
```suggestion
    const auto it = fd_entry_.find(sockfd);
    if (it == fd_entry_.cend()) {
```
find returns an iterator which is usually copied instead of referenced. Common practice is to name it `it` in a short scope, but this is optional.